src/app/guards/valid-user.guard.ts

```diff
@@ -2,14 +2,14 @@ import { Injectable } from '@angular/core';
 import { CanActivate, Router } from '@angular/router';
 
 @Injectable()
-export class AuthCheckGuard implements CanActivate {
+export class ValidUserGuard implements CanActivate {
```

Comment:
AuthCheckGuard was better itself. I suggest to rename to AuthRequiredGuard

Reply:
I think `AuthRequiredGuard` may not be that much intuitive here because the guard will return true for anonymous user also, and we already have `AuthGuard` for checking authentication.

Reply:
AuthRequiredGuard should return false if user can access anonymously or already authenticated and has token; 


If auth is mandatory (required) and no token, then return true.